CONTRIBUTING.md

```diff
@@ -0,0 +1,128 @@
+    - [Cleaning up commits](#cleaning-up-commits)
+    - [Breaking changes](#breaking-changes)
+    - [General guidelines](#general-guidelines)
+    - [Testing guidelines](#testing-guidelines)
+    - [SDK generation guidelines](#sdk-generation-guidelines)
+
+## Before Starting
+
+### Onboarding
+
+Make sure that your GitHub account is part of the Azure organization. [Use this page](http://aka.ms/azuregithub) to link your account.
+
+Before cloning this repository, please make sure you have started in our [documentation repository](https://github.com/Azure/adx-documentation-pr) (you will only have access to that page if you are part of the Azure organization).
+
+### GitHub Basics
+
+#### GitHub Workflow
+
+If you don't have experience with Git and GitHub, some of the terminology and process can be confusing. [Here is a guide to understanding the GitHub flow](https://guides.github.com/introduction/flow/) and [here is a guide to understanding the basic Git commands](https://services.github.com/kit/downloads/github-git-cheat-sheet.pdf).
+
+### Forking the Azure/azure-sdk-for-net repository
+
+Unless you are working with multiple contributors on the same file, we ask that you fork the repository and submit your pull request from there. [Here is a guide to forks in GitHub](https://guides.github.com/activities/forking/).
+
+###  Code of Conduct
+
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+## Filing Issues
+
+You can find all of the issues that have been filed in the [Issues](https://github.com/Azure/azure-sdk-for-net/issues) section of the repository.
+
+If you encounter any bugs with Azure SDK for .NET, or would like to request a feature, please file an issue [here](https://github.com/Azure/azure-sdk-for-net/issues/new).
+
+You can find the dates for the next three sprints for Azure SDK for .NET in the [Milestones](https://github.com/Azure/azure-sdk-for-net/milestones) section of the Issue page. Each milestone will display the issues that are being worked on for each corresponding sprint.
+
+## Making Changes
+
+### Pull Requests
+
+You can find all of the pull requests that have been opened in the [Pull Request](https://github.com/Azure/azure-sdk-for-net/pulls) section of the repository.
+
+To open your own pull request, click [here](https://github.com/Azure/azure-sdk-for-net/compare). When creating a pull request, keep the following in mind:
+- Make sure you are pointing to the fork and branch that your changes were made in
+- Choose the correct branch you want your pull request to be merged into
+    - The **AutoRest** branch is for code generated from AutoRest
+    - The **master** branch is for code generated from Hyak
```

Comment:
also for **master** branch please state that code there is in support mode only.